File: src/main/webapp/js/service/categoryService.js
/*定义服务层*/
app.service("categoryService",function ($http) {

    this.findAll = function () {
        return $http.get("/category/findAll");
    }

    this.findOne = function (id) {
        return $http.get("/category/findOne?id="+id);
    }

    this.delete = function (selectedIds) {
        return $http.post("/category/delete",selectedIds);
    }

    this.update = function (entity) {
        return $http.post("/category/update",entity);
    }

    this.add = function (entity) {
        return $http.post("/category/add",entity);
    }

});

File: src/main/java/com/wangkaihua/myblog/service/CategoryService.java
package com.wangkaihua.myblog.service;

import com.wangkaihua.myblog.entity.TbCategory;

import java.util.List;

/**
 * @author wangkaihua
 * @create 2018-09-04 16:25
 **/
public interface CategoryService {
    List<TbCategory> findAll();

    void add(TbCategory category);

    void update(TbCategory category);

    TbCategory findOne(int id);

    void delete(int[] ids);
}
